modules/olmv1-updating-an-operator.adoc

```diff
@@ -0,0 +1,123 @@
+// Module included in the following assemblies:
+//
+// * operators/olm_v1/olmv1-installing-an-operator-from-a-catalog.adoc
+
+:_content-type: PROCEDURE
+
+[id="olmv1-updating-an-operator_{context}"]
+= Updating an Operator
+
+You can update your Operator by manually editing your Operator's custom resource (CR) and applying the changes.
+
+.Prerequisites
+
+* You have a catalog installed.
+* You have an Operator installed.
+
+.Procedure
+
+. Inspect your Operator's package contents to find which channels and versions are available for updating by running the following command:
++
+[source,terminal]
+----
+$ oc get package <catalog_name>-<operator_name> -o yaml
+----
++
+.Example command
+[source,terminal]
+----
+$ oc get package redhat-operators-quay-operator -o yaml
+----
+
+. Edit your Operator's CR to update the version to `3.9.1`, as shown in the following example:
++
+.Example `test-operator.yaml` CR
+[source,yaml]
+----
+apiVersion: operators.operatorframework.io/v1alpha1
+kind: Operator
+metadata:
+  name: quay-example
+spec:
+  packageName: quay-operator
+  version: 3.9.1 <1>
+----
+<1> Update the version to `3.9.1`
+
+. Apply the update to the cluster by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f test-operator.yaml
+----
++
+.Example output
+[source,text]
+----
+operator.operators.operatorframework.io/quay-example configured
+----
++
+[TIP]
+====
+You can patch and apply the changes to your Operator's channel and version from the CLI by running the following command:
+
+[source,terminal]
+----
+$ oc patch operator.operators.operatorframework.io/quay-example -p \
```

Comment:
@michaelryanpeter following the installation, if you update it with only updating version (not removing the channel), the command will does not work
```console
[root@preserve-olm-env2 olmv1review]# oc patch operator.operators.operatorframework.io/quay-example -p \
>   '{"spec":{"version":"3.9.1"}}' \
>   --type=merge
operator.operators.operatorframework.io/quay-example patched
[root@preserve-olm-env2 olmv1review]# oc get operator.operators.operatorframework.io/quay-example -o yaml
apiVersion: operators.operatorframework.io/v1alpha1
kind: Operator
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"operators.operatorframework.io/v1alpha1","kind":"Operator","metadata":{"annotations":{},"name":"quay-example"},"spec":{"channel":"stable-3.8","packageName":"quay-operator","version":"3.8.12"}}
  creationTimestamp: "2023-10-19T02:42:21Z"
  generation: 2
  name: quay-example
  resourceVersion: "140524"
  uid: 5b3895d4-e008-42ee-8e45-84ce3c31bcad
spec:
  channel: stable-3.8
  packageName: quay-operator
  version: 3.9.1
status:
  conditions:
  - lastTransitionTime: "2023-10-19T02:43:08Z"
    message: package 'quay-operator' at version '3.9.1' in channel 'stable-3.8' not
      found
    observedGeneration: 2
    reason: ResolutionFailed
    status: "False"
    type: Resolved
  - lastTransitionTime: "2023-10-19T02:43:08Z"
    message: installation has not been attempted as resolution failed
    observedGeneration: 2
    reason: InstallationStatusUnknown
    status: Unknown
    type: Installed
```

so, for such update, you have to set channel because the installation section has channel.
so, I suggest to keep channel. if you want to remove channel still, please remove this patch method.

by the way, if you apply yam file which only have version, no such issue because the channel is removed.

Reply:
@kuiwang02 I have updated the commands and example outputs to use only the target version. Thank you for pointing out my error.